### PR TITLE
made descriptions optional 

### PR DESCRIPTION
**Pull request by billybobjoe123**
#11 
made descriptions optional in form and controller. edited frontend form test to not require description.
deployment link: https://proj-courses-dev-billybobjoe123.dokku-02.cs.ucsb.edu/

### CODE REVIEW

**Comment by berdly** [comment]
Looks good mostly although there's a few things I'm noticing: 

- It's redundant to set required to false on the `RequestParam` annotation since the argument is now an `Optional<T>` it's like putting `Optional<Optional<T>>`. You should probably remove the `required=false` bit since Spring 'understands' what we're trying to say with the typing alone.

- Since `PersonalSchedule.description` is always optional now, we should consider changing this variable's type to `Optional<String>` instead of simply making it empty when there's no value. That would probably need to be it's own issue though since it would require changes in a decent amount of places. If anyone with authority who's reading this could weigh in, I'd be interested to see if that would be considered a contribution worth points

**Comment by berdly** [comment]
Sorry but the `RequestParam` annotation is still needed, just not the `(required = false)` part.

**Review by berdly — APPROVED** [review]

File: frontend/src/tests/components/PersonalSchedules/PersonalScheduleForm.test.js [hunk 1]
```diff
@@ -81,7 +81,7 @@ describe("PersonalScheduleForm tests", () => {
     fireEvent.click(submitButton);
 
     expect(await screen.findByText(/Name is required./)).toBeInTheDocument();
-    expect(screen.getByText(/Description is required./)).toBeInTheDocument();
+    //expect(screen.getByText(/Description is required./)).toBeInTheDocument();    no longer necessary due to issue #11 making descriptions optional
```
Comment (pconrad) [inline frontend/src/tests/components/PersonalSchedules/PersonalScheduleForm.test.js:84]: Please do not leave commented out code in the PR.  Remove it.  

Here's why:
https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html#why-should-commented-out-code-be-removed

**Review by pconrad — CHANGES_REQUESTED** [review]
Apart from the [commented out code](https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html#why-should-commented-out-code-be-removed) LGTM.  Fix that and then it's mergeable.

**Comment by billybobjoe123** [comment]
deployment link:
https://proj-courses-dev-billybobjoe123.dokku-02.cs.ucsb.edu/

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**